react-components/src/query/network/searchClassicAssetsForModels.test.ts

```diff
@@ -0,0 +1,177 @@
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+import { searchClassicAssetsForModels } from './searchClassicAssetsForModels';
+import { cadModelOptions } from '#test-utils/fixtures/cadModel';
+import { Mock, It } from 'moq.ts';
+import {
+  type Asset,
+  type AssetMapping3D,
+  type AssetMappings3DAPI,
+  type AssetsAPI,
+  type CogniteClient,
+  type CursorAndAsyncIterator,
+  type HttpResponse,
+  type IdEither
+} from '@cognite/sdk';
+import { createAssetMock } from '#test-utils/fixtures/assets';
+import assert from 'assert';
+import { isInternalId } from '../../utilities/instanceIds';
+import { type AssetMappingAndNode3DCache } from '../../components/CacheProvider/AssetMappingAndNode3DCache';
+import { type RevealRenderTarget } from '../../architecture';
+import { type CdfAssetMapping } from '../../components/CacheProvider/types';
+
+const ARBITRARY_SEARCH_LIMIT = 100;
+const ARBITRARY_NODE_ID = 200;
+const ARBITRARY_TREE_INDEX = 143;
+
+const TEST_ASSETS = [createAssetMock(0), createAssetMock(1), createAssetMock(2)];
+const TEST_PROJECT = 'test_project';
+
+const mockAssetMappings3dFilter = vi.fn<AssetMappings3DAPI['filter']>();
+const mockAssetsRetrieve = vi.fn<AssetsAPI['retrieve']>();
+const mockPostAssetList = vi.fn<CogniteClient['post']>();
+
+const mockSdk = new Mock<CogniteClient>()
+  .setup((p) => p.assetMappings3D.filter)
+  .returns(mockAssetMappings3dFilter)
+  .setup((p) => p.assets.retrieve)
+  .returns(mockAssetsRetrieve)
+  .setup((p) => p.project)
+  .returns(TEST_PROJECT)
+  .setup(
+    async (p) =>
+      await p.post(
+        It.Is((url) => url === `/api/v1/projects/${TEST_PROJECT}/assets/list`),
+        It.IsAny()
+      )
+  )
+  .callback(async ({ args: [url, parameters] }) => await mockPostAssetList(url, parameters))
+  .object();
+
+const mockGetAssetMappingsForModel =
+  vi.fn<AssetMappingAndNode3DCache['getAssetMappingsForModel']>();
+
+const mockRenderTarget = new Mock<RevealRenderTarget>()
+  .setup((p) => p.cdfCaches.assetMappingAndNode3dCache)
+  .returns(
+    new Mock<AssetMappingAndNode3DCache>()
+      .setup((p) => p.getAssetMappingsForModel)
+      .returns(mockGetAssetMappingsForModel)
+      .object()
+  )
+  .object();
+
+describe(searchClassicAssetsForModels.name, () => {
+  beforeEach(() => {
+    vi.resetAllMocks();
+    mockAssetMappings3dFilter.mockImplementation(
+      async () =>
+        await (Promise.resolve({ items: [] }) as unknown as CursorAndAsyncIterator<AssetMapping3D>)
```

Comment:
I have been unable to mock the `CursorAndAsyncIterator` return value from the SDK without resorting to `as unknown as CursorAndAsyncIterator`. I'd be glad to hear opinions on how to solve this better